app/models/ems_refresh/refreshers/container_refresher_mixin.rb

```diff
@@ -0,0 +1,44 @@
+module EmsRefresh
+  module Refreshers
+    module ContainerRefresherMixin
+      FETCH_ENTITIES = {
+        :kubernetes => [
+          {:name => 'pods'}, {:name => 'services'}, {:name => 'replication_controllers'}, {:name => 'nodes'},
+          {:name => 'endpoints'}, {:name => 'namespaces'}, {:name => 'resource_quotas'}, {:name => 'limit_ranges'},
+          {:name => 'persistent_volumes'}, {:name => 'persistent_volume_claims'},
+          # workaround for: https://github.com/openshift/origin/issues/5865
+          {:name => 'component_statuses', :default => []}
+        ],
+        :openshift  => [
+          {:name => 'routes'}, {:name => 'projects'}
+        ]
+      }
+      def kubernetes_parse_inventory(kubeclient)
+        entities = fetch_entities(kubeclient, FETCH_ENTITIES[:kubernetes])
+        EmsRefresh.log_inv_debug_trace(entities, "inv_hash:")
+        ManageIQ::Providers::Kubernetes::ContainerManager::RefreshParser.ems_inv_to_hashes(entities)
+      end
+
+      def openshift_parse_inventory(openshift_client, kubeclient)
+        entities =
+        fetch_entities(openshift_client, FETCH_ENTITIES[:openshift]).merge(
+          fetch_entities(kubeclient, FETCH_ENTITIES[:kubernetes])
+        )
```

Comment:
Doing that will entail changing how we log.

I will have to do logging in parse_inventory after the merge, but I can't since inv has already turned to hash.